packages/openapi-v3/src/decorators/request-body.spike.decorator.ts

```diff
@@ -0,0 +1,118 @@
+
+const debug = require('debug')('loopback:openapi3:metadata:requestbody');
+
+function isRequestBodySpec<T extends object>(spec: Partial<RequestBodyObject> | JsonSchemaOptions<T>): spec is Partial<RequestBodyObject> {
+  return (spec as Partial<RequestBodyObject>).description !== undefined
+    || (spec as Partial<RequestBodyObject>).required !== undefined
+    || (spec as Partial<RequestBodyObject>).content !== undefined;
+}
+
+export function requestBody2<T extends object>(
+  specOrModelOrOptions?: Partial<RequestBodyObject> | Function | JsonSchemaOptions<T>,
+  modelOrOptions?: Function | JsonSchemaOptions<T>,
+  schemaOptions?: JsonSchemaOptions<T>
+) {
+  if (typeof specOrModelOrOptions == 'function') {
+    // omit the 1st param
+    // @requestBody(modelCtor, schemaOptions)
+    if (modelOrOptions && typeof modelOrOptions == 'object')
+      return _requestBody2({}, specOrModelOrOptions, modelOrOptions as JsonSchemaOptions<T>);
+    // @requestBody(modelCtor)
+    return _requestBody2({}, specOrModelOrOptions);
+  } else if (specOrModelOrOptions && isRequestBodySpec(specOrModelOrOptions)) {
+    // 1st param is spec
+    if (modelOrOptions && typeof modelOrOptions == 'object')
+      // omit the 2nd param
+      // @requestBody(spec, schemaOptions)
+      return _requestBody2(specOrModelOrOptions, undefined, modelOrOptions)
+    // @requestBody(spec)
+    // @requestBody(spec, modelCtor)
+    // @requestBody(spec, modelCtor, schemaOptions)
+    return _requestBody2(specOrModelOrOptions, modelOrOptions, schemaOptions);
+  } else if (specOrModelOrOptions !== undefined) {
+    // omit 1st and 2nd params
+    // @requestBody(schemaOptions)
+    return _requestBody2({}, undefined, specOrModelOrOptions as JsonSchemaOptions<T>);
+  } else {
+    // no params
+    // @requestBody()
+    return _requestBody2();
+  }
+}
+
+// `name` is provided to avoid generating the same schema
+function _requestBody2<T extends object>(
+  requestBodySpecification?: Partial<RequestBodyObject>,
+  modelCtor?: Function,
+  schemaOptions?: JsonSchemaOptions<T>
+) {
+  return function (target: object, member: string, index: number) {
+    debug('@newRequestBody() on %s.%s', target.constructor.name, member);
+    debug('  parameter index: %s', index);
+    /* istanbul ignore if */
+    if (debug.enabled)
+      debug('  schemaOptions: %s', inspect(requestBodySpecification, { depth: null }));
+
+    // Use 'application/json' as default content if `requestBody` is undefined
+    const requestBodySpec = Object.assign({}, requestBodySpecification || { content: {} });
+
+    if (_.isEmpty(requestBodySpec.content))
+      requestBodySpec.content = { 'application/json': {} };
+
+    // Get the design time method parameter metadata
+    const methodSig = MetadataInspector.getDesignTypeForMethod(target, member);
+    const paramTypes = (methodSig && methodSig.parameterTypes) || [];
+
+    const paramType = paramTypes[index];
+
+    // preserve backward compatibility
```

Comment:
FYI: compared with the current decorator function, the difference if from Line 81 to Line 91, other code stay the same.